### PR TITLE
Refactor most of knight logic back into chess_piece.rb

### PR DESCRIPTION
**Pull request by radiofreemattd**
I think this is essentially what @jonmast was asking for in his initial comment on #40.

### CODE REVIEW

**Comment by radiofreemattd** [comment]
I think the concept of calculating x and y movement is pretty basic to *all* pieces, which suggests that `chess_piece.rb` is the right place for that logic. Plus, it simplifies some of the logic in `#is_diagonal?`, which is what I think @jonmast was originally after in #40 as I mentioned above.

The only thing that I don't like is that the actual code in `knight.rb` is a little less explicit in its intent - you have to rely on an understanding of the inheritance model (not particularly complex but maybe just less than ideal).

**Comment by jonmast** [comment]
:+1: Looks good to me

File: app/models/chess_piece.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Chess piece model
 class ChessPiece < ActiveRecord::Base
-  BOARD_START_INDEX = 1
+  BOARD_BGN_INDEX = 1
```
Comment (mscoutermarsh) [inline app/models/chess_piece.rb]: I try to not use acronyms. What's a BGN?

Comment (radiofreemattd) [inline app/models/chess_piece.rb]: Begin...just was hoping to get the constants to line up nicely...not a big deal. :+1:

**Pull request merged**